api/src/main/java/org/pmiops/workbench/google/DirectoryService.java

```diff
@@ -1,25 +1,29 @@
 package org.pmiops.workbench.google;
 
 import com.google.api.services.directory.model.User;
+import java.util.Optional;
 
 /** Encapsulate Googe APIs for handling GSuite user accounts. */
 public interface DirectoryService {
   boolean isUsernameTaken(String username);
 
-  /** Returns a user via email address lookup. */
+  /** Returns a user via email address lookup. Returns null if no user was found. */
   User getUser(String email);
 
   /**
    * Returns a user via username lookup (e.g. the user's GSuite email address without the domain
-   * suffix.
+   * suffix. Returns null if no user was found.
    */
   User getUserByUsername(String username);
 
+  /** Looks up a user by username and returns their stored contact email address, if available. */
+  Optional<String> getContactEmail(String username);
+
   /**
-   * Looks up a user by username and returns their stored contact email address. If no contact email
-   * is stored in G Suite, then null is returned.
+   * Looks up a user by GSuite email address and returns their stored contact email address, if
+   * available.
    */
-  String getContactEmailAddress(String username);
+  Optional<String> getContactEmailFromGSuiteEmail(String gSuiteEmailAddress);
```

Comment:
this is the only place I've seen `gSuiteEmailAddress` instead of `username`. Is the idea here that there may not actually be a user yet?

Reply:
Mostly for consistency and clarity within this file, since `username` within this service has an existing meaning (which is the GSuite email without the domain suffix). It looks like it would be possible to drop that meaning from this service – IMO not a huge win, I'll opt to leave as-is.

Reply:
In that case, I'd rename `username` in this class then to `usernamePrefix` or some such.

FWIW I rarely absorb code a file at a time. It's 80% hopping via IDE linkage between occurrences. Email already means two things. If username means two things, then we're in trouble.

Reply:
I was about to punt on this, but when I looked through even this file I found there was some inconsistency in how `username` alone was being used. I took a broader pass at cleaning up the naming in existing methods too (turns out most of these are only used in one place, so the surface area was relatively small). Feels like an improvement, as long as I didn't break anything. PTAL.